### PR TITLE
UseWithComponent (Functionally generic version of LogComponent)

### PR DESCRIPTION
**Pull request by HoofedEar**
## About the PR 
Basically, adds a component that allows you to define an acceptable tag for producing something from an item.
(think of using a hatchet on towercap logs, or cutting pizza slices with a knife.)

**Screenshots**

image

**Changelog**

N/A - Adds component for ppl to use

### CODE REVIEW

File: Content.Server/UseWith/UseWithComponent.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+[RegisterComponent]
+[Friend(typeof(UseWithSystem))]
+public sealed class UseWithComponent : Component
+{
+    [DataField("spawnedPrototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string? SpawnedPrototype;
+
+    [DataField("spawnCount")] 
+    public int SpawnCount;
+
```
Comment (mirrorcult) [inline Content.Server/UseWith/UseWithComponent.cs:17]: this needs to use a `List<EntitySpawnEntry>`

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/UseWith/UseWithComponent.cs (hunk 1 above)
Comment (HoofedEar) [reply Content.Server/UseWith/UseWithComponent.cs:17]: Oh duh good call! Allows for even random results!!

**Review by moonheart08 — CHANGES_REQUESTED** [review]
This is dead code and cannot be added without some kind of usage.
I'd advise adding the log interaction you mentioned.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Doesn't this duplicate mineablecomponent? Also needs some kind of usage + documentation as moony said please.

**Comment by HoofedEar** [comment]
After further research, this actually won't be useful. pizza slices do their own thing (along with accounting for nutrition of slices) and Log component cares about something having the Sharp component so it would require updating a lot of prototypes.

**Pull request closed without merging**